### PR TITLE
Remove inline js from ia_thirdparty_logins.html

### LINKED ISSUE #8379: Remove inline js: account/ia_thirdparty_logins.html
Remove the inline JS from this file following the recommended steps in #4474

### Proposal & Constraints

This one should be relatively straightforward.

### Related files

https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/account/ia_thirdparty_logins.html

### PR DESCRIPTION
**Pull request by AdwaitSalankar**
Closes #8379 

Removes inline js from ia_thirdparty_logins.html and moves it to ia_thirdparty_logins.js.

Also,
- Changed `==` to `===` 
- Changed `var` to `const`
- Removed unneeded `\` from regex
- Changed the string to template literal.

in the `handleMessageEvent` function 

### Stakeholders
@RayBB

### CODE REVIEW

**Comment by AdwaitSalankar** [comment]
@RayBB Can you review this?

File: openlibrary/plugins/openlibrary/js/index.js [hunk 1]
```diff
@@ -547,4 +547,8 @@ jQuery(function () {
             .then(module => module.initBreadcrumbSelect(crumbs));
     }
 
+    if (document.getElementById('ia-third-party-logins')) {
```
Comment (RayBB) [inline openlibrary/plugins/openlibrary/js/index.js:550]: Can we do something like this so we only do the query once and pass it in?
https://github.com/internetarchive/openlibrary/pull/8374/files#diff-587fb865c85bb3325504c795a40db189d2e702d42d9d160304607e027c142ebd

**Review by RayBB — CHANGES_REQUESTED** [review]
Generally looks okay just one suggestion. We'll need a review from a staff member to merge it. Maybe @jimchamp who has reviewed a few of these.

File: openlibrary/plugins/openlibrary/js/index.js (hunk 1 above)
Comment (AdwaitSalankar) [reply openlibrary/plugins/openlibrary/js/index.js:550]:
```js
const thirdPartyLoginsIframe = document.getElementById('ia-third-party-logins');
```
Does the name `thirdPartyLoginsIframe` look good?

**Comment by AdwaitSalankar** [comment]
@RayBB I have made the changes.

**Review by RayBB — APPROVED** [review]
Everything looks good to me. We'll need someone else to give a final check and merge.
@jimchamp maybe can help or tag someone who can

**Comment by AdwaitSalankar** [comment]
Hi @jimchamp, can you review this PR?

**Comment by AdwaitSalankar** [comment]
@cdrini Can you review this PR?

**Comment by cdrini** [comment]
Hi @AdwaitSalankar ! Just a heads up , this one will take a little longer for us to get to because it touches the login pages which we can only fully test on staging/production, which requires more time from us to get through. Feel free to tackle another issue in the meantime!

**Review by jimchamp — APPROVED** [review]
Thanks for helping with this, @AdwaitSalankar!  Sorry that it took so long to review....

Everything seems to be working as expected.  Was able to link an existing archive account to an OL account and login in `testing`.

**Pull request merged**